### PR TITLE
Setup release script and gradle release / publish tasks

### PR DESCRIPTION
**Pull request by etanshaul**
progress towards #87 

Missing from #87 is the backend kokoro config which is done separately (not in this repo).

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -79,3 +82,20 @@ dependencies {
...
+}
+
+release {
+    tagTemplate = "v\${version}"
+
+    val git: GitAdapter.GitConfig = getProperty("git") as GitAdapter.GitConfig
+    git.requireBranch = "/^release_v\\d+.*$/"
+}
+
+inline operator fun <T : Task> T.invoke(a: T.() -> Unit): T = apply(a)
```
Comment (ivanporty) [inline build.gradle.kts]: Interesting, we convert apply to direct call for Task?

File: build.gradle.kts [hunk 2]
```diff
@@ -79,3 +82,20 @@ dependencies {
 
     compile("com.google.protobuf:protobuf-java:2.5.0")
 }
+
+val intellijRepoChannel: String by project
+val publishPlugin: PublishTask by tasks
+publishPlugin {
```
Comment (ivanporty) [inline build.gradle.kts]: So as I understand, IJ release process uses `net.researchgate.release`?

File: build.gradle.kts (hunk 1 above)
Comment (etanshaul) [reply build.gradle.kts]: Yeah. Makes the task definitions a bit cleaner

File: build.gradle.kts (hunk 2 above)
Comment (etanshaul) [reply build.gradle.kts]: right. that is the gradle release plugin, called by `gradlew release`. This does some tagging, updates the version in `settings.gradle` to the next version, pushes a PR to github. 

To do the actual release, we need to run the Kokoro release job (not yet setup for this repo). This release job will invoke the release.sh script in the PR. (previously when we were using Jenkins we had the ability to configure Kokoro to detect a "release" triggered from the gradle release plugin and automatically call the release job. But it wasn't possible with Kokoro last I checked).

**Pull request merged**